pandora-common/src/networking/directory_client.ts

```diff
@@ -101,6 +101,40 @@ export const ACCOUNT_SETTINGS_DEFAULT = Object.freeze<IDirectoryAccountSettings>
 	interfaceChatroomOfflineCharacterFilter: 'ghost',
 });
 
+export const DirectoryAccountSettingsLimitedSchema = z.object({
+	displayName: DisplayNameSchema,
+});
+export type IDirectoryAccountSettingsLimited = z.infer<typeof DirectoryAccountSettingsLimitedSchema>;
+
+type DirectoryAccountSettingsLimitedStoredSchemaAcc = {
+	[K in keyof IDirectoryAccountSettingsLimited]: ZodObject<{
+		value: typeof DirectoryAccountSettingsLimitedSchema.shape[K];
+		nextAllowedChange: ZodNumber;
+	}>;
+};
+export const DirectoryAccountSettingsLimitedStoredSchema = z.object(
+	KnownObject.entries(DirectoryAccountSettingsLimitedSchema.shape).reduce(
```

Comment:
Avoid using `reduce` where the reduced value actually never changes - using a much simpler `for` will have almost identical code.

Reply:
I have no idea why this would be beneficial, considering that we would need the for to be in global scope or an immediately invoked lambda, non of which seems better for me